app/javascript/dashboard/composables/useFontSize.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * @file useFontSize.js
+ * @description A composable for managing font size settings throughout the application.
+ * This handles font size selection, application to the DOM, and persistence in user settings.
+ */
+
+import { computed, watch } from 'vue';
+import { useUISettings } from 'dashboard/composables/useUISettings';
+import { useAlert } from 'dashboard/composables';
+import { useI18n } from 'vue-i18n';
+
+/**
+ * Font size options with their pixel values
+ * @type {Object}
+ */
+const FONT_SIZE_OPTIONS = {
+  SMALLER: '14px',
+  SMALL: '15px',
+  DEFAULT: '16px',
+  LARGE: '18px',
+  LARGER: '20px',
+  EXTRA_LARGE: '22px',
+};
+
+/**
+ * Array of font size option keys
+ * @type {Array<string>}
+ */
+const FONT_SIZE_NAMES = Object.keys(FONT_SIZE_OPTIONS);
+
+/**
+ * Font size management composable
+ *
+ * @returns {Object} Font size utilities and state
+ * @property {Array} fontSizeOptions - Array of font size options for select components
+ * @property {import('vue').ComputedRef<string>} currentFontSize - Current font size from UI settings
+ * @property {Function} applyFontSize - Function to apply font size to document
+ * @property {Function} updateFontSize - Function to update font size in settings with alert feedback
+ */
+export const useFontSize = () => {
+  const { uiSettings, updateUISettings } = useUISettings();
+  const { t } = useI18n();
+
+  /**
+   * Font size options for select dropdown
+   * @type {Array<{value: string, label: string}>}
+   */
+  const fontSizeOptions = FONT_SIZE_NAMES.map(name => ({
+    value: FONT_SIZE_OPTIONS[name],
+    label: t(
+      `PROFILE_SETTINGS.FORM.INTERFACE_SECTION.FONT_SIZE.OPTIONS.${name}`
+    ),
+  }));
+
+  /**
+   * Current font size from UI settings
+   * @type {import('vue').ComputedRef<string>}
+   */
+  const currentFontSize = computed(
+    () => uiSettings.value.font_size || FONT_SIZE_OPTIONS.DEFAULT
+  );
+
+  /**
+   * Apply font size to document root
+   *
+   * @param {string} pixelValue - Font size in pixels (e.g., '16px')
+   * @returns {void}
+   */
+  const applyFontSize = pixelValue => {
+    // Use requestAnimationFrame for better performance
+    requestAnimationFrame(() => {
+      if (!pixelValue || pixelValue === FONT_SIZE_OPTIONS.DEFAULT) {
+        document.documentElement.style.removeProperty('font-size');
+      } else {
+        document.documentElement.style.setProperty('font-size', pixelValue);
+      }
+    });
```

Comment:
```suggestion
    requestAnimationFrame(() => {
      document.documentElement.style.setProperty('font-size', pixelValue ?? FONT_SIZE_OPTIONS.DEFAULT);
    });
```

@iamsivin WDYT?

Reply:
@scmmishra If we want to set the default 16px all the time then, I guess we can do this, else the existing approach is better. It will not set the `font-size` property for default case.

Reply:
Should be fine I guess, what is our tailwind default? If it's 16, we're good

Reply:
@scmmishra Updated the PR https://github.com/chatwoot/chatwoot/pull/10974/commits/31f537fae01575ca4608b03beb4b0203fc729b63